source/agora/consensus/validation/Block.d

```diff
@@ -113,6 +119,282 @@ public string isInvalidReason (const ref Block block, in ulong prev_height,
     return null;
 }
 
+/*******************************************************************************
+
+    Check the validity of a genesis block
+
+    Follow the Block rules except for the following
+        - Block height is 0
+        - The previous block hash of the block must be empty
+        - The number of transactions in the block is above zero and less than
+          or equal to Block.TxsInBlock
+        - All the transactions pass validation, which implies:
+            - Transaction type in block should only be Payment or Freeze
+            - The inputs of the transaction(s) are empty
+            - There is one or more outputs of the transaction
+        - All the enrollments pass validation, which implies:
+            - The enrollments refer to freeze tx's in this block
+
+    Params:
+        block = The genesis block to check
+
+    Returns:
+        `null` if the genesis block is valid, otherwise a string explaining
+        the reason it is invalid.
+
+*******************************************************************************/
+
+public string isValidGenesisBlock (immutable Block* block) nothrow @safe
+{
+    if (block.header.height != 0)
+        return "GenesisBlock: The height of the block is not 0";
+
+    if (block.header.prev_block != Hash.init)
+        return "GenesisBlock: Header.prev_block is not empty";
+
+    if (block.txs.length == 0)
+        return "GenesisBlock: Transaction(s) are empty";
+
+    if (!block.txs.isSorted())
```

Comment:
I asked to move this below the length check. But it seems my comment is gone?

Reply:
Could you give me your opinion one more time?

Reply:
https://github.com/bpfkorea/agora/pull/884#discussion_r435957214

You check the length, then the is sorted call, then the length again. The sorted call should be after all the length calls.

Reply:
I apply the suggestion.